packages/reputation-miner/test/MaliciousReputationMinerWrongJRH.js

```diff
@@ -0,0 +1,67 @@
+import ReputationMiner from "../ReputationMiner";
+
+const ethers = require("ethers");
+
+class MaliciousReputationMinerWrongJRH extends ReputationMiner {
+  // Only difference between this and the 'real' client should be that it submits a bad JRH
+
+  constructor(opts, entryToFalsify) {
+    super(opts);
+    this.entryToFalsify = entryToFalsify.toString();
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async respondToChallenge() {
+    // This client sometimes won't be able to respond to challenge - we mess up its JRH with a hash it doesn't know about
+  }
+
+  async submitRootHash(startIndex = 1) {
+    const hash = await this.getRootHash();
+    const repCycle = await this.getActiveRepCycle();
+    // Get how much we've staked, and thefore how many entries we have
+    let entryIndex;
+    const [, balance] = await this.tokenLocking.getUserLock(this.clnyAddress, this.minerAddress);
+    const reputationMiningWindowOpenTimestamp = await repCycle.getReputationMiningWindowOpenTimestamp();
+    const minStake = ethers.utils.bigNumberify(10).pow(18).mul(2000); // eslint-disable-line prettier/prettier
+    for (let i = ethers.utils.bigNumberify(startIndex); i.lte(balance.div(minStake)); i = i.add(1)) {
+      // Iterate over entries until we find one that passes
+      const entryHash = await repCycle.getEntryHash(this.minerAddress, i, hash); // eslint-disable-line no-await-in-loop
+
+      const miningCycleDuration = 60 * 60 * 24;
```

Comment:
Can we import this from `constants.js`?

Reply:
I think I'd rather not. We have the duration in `ReputationMiner.js` (i.e. the 'real' miner) too, and I wouldn't want to change it there either - I don't want this package repository (which will probably eventually be published as its own npm package) to be reliant on the whole of `colonyNetwork`.

Reply:
Fair enough